src/main/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethods.groovy

```diff
@@ -164,16 +166,40 @@ class RoddyIOHelperMethods {
 
     /**
      * This method copies a file and tries to preserve access rights using Java get/set rights methods.
+     *
+     * Modified after http://docs.oracle.com/javase/8/docs/api/.
+     *
      * @param src
      * @param tgt
      */
-    public static void copyDirectory(File src, File tgt) {
+    static void copyDirectory(File source, File target, CopyOption... options = [StandardCopyOption.COPY_ATTRIBUTES]) {
```

Comment:
So much code :(. I've seen some examples, so you did everything right. One idea I just had was to maybe use
dir.eachFileRecurse (FileType.FILES) { file ->  }
from Groovy to make things simpler. But if it works, just leave it!